cli/cook/subcommands/kill.py

```diff
@@ -0,0 +1,109 @@
+            bullets = []
+            for entry in uuid_to_entries[duplicate_uuid]:
+                bullets.append(f'- as a {entry["type"]} on {entry["cluster_name"]}')
+            message = f'{duplicate_uuid} is duplicated:\n' + '\n'.join(bullets)
+            messages.append(message)
+        details = '\n\n'.join(messages)
+        message = f'Unable to kill due to duplicate UUIDs.\n\n{details}'
+        raise Exception(message)
+
+
+def kill_entities(uuids, query_result, clusters):
+    """Attempts to kill the jobs / instances / groups with the given UUIDs"""
+    exit_code = 0
+    for cluster_name, entities in query_result['clusters'].items():
+        cluster = next(c for c in clusters if c['name'] == cluster_name)
+
+        jobs = entities['jobs']
+        instances = [i for j in entities['instances'] for i in j['instances'] if i['task_id'] in uuids]
+        if len(jobs) > 0 or len(instances) > 0:
+            job_uuids = [j['uuid'] for j in jobs]
+            instance_uuids = [i['task_id'] for i in instances]
+            resp = http.delete(cluster, 'rawscheduler', params={'job': job_uuids, 'instance': instance_uuids})
+            if resp.status_code == 204:
+                for job in jobs:
+                    print_info(f'Killed job {colors.bold(job["uuid"])} on {colors.bold(cluster_name)}.')
+                for instance in instances:
+                    instance_uuid = instance['task_id']
+                    print_info(f'Killed job instance {colors.bold(instance_uuid)} on {colors.bold(cluster_name)}.')
+            else:
+                exit_code = 1
+                for job in jobs:
+                    print(colors.failed(f'Failed to kill job {job["uuid"]} on {cluster_name}.'))
+                for instance in instances:
+                    print(colors.failed(f'Failed to kill job instance {instance["task_id"]} on {cluster_name}.'))
+
+        groups = entities['groups']
+        if len(groups) > 0:
+            group_uuids = [g['uuid'] for g in groups]
+            resp = http.delete(cluster, 'group', params={'uuid': group_uuids})
+            if resp.status_code == 204:
+                for group in groups:
+                    print_info(f'Killed job group {colors.bold(group["uuid"])} on {colors.bold(cluster_name)}.')
+            else:
+                exit_code = 1
+                for group in groups:
+                    print(colors.failed(f'Failed to kill job group {group["uuid"]} on {cluster_name}.'))
+
+    return exit_code
+
+
+def kill(clusters, args):
+    """Attempts to kill the jobs / instances / groups with the given UUIDs."""
+    uuids = strip_all(args.get('uuid'))
+    query_result = query(clusters, uuids)
+    if query_result['count'] == 0:
+        print_no_data(clusters)
+        return 1
+
+    # If the user provides UUIDs that map to more than one entity,
+    # we will raise an Exception that contains the details
+    guard_against_duplicates(uuids, query_result)
```

Comment:
I think it would be better to catch the exception thrown in this method, print the error message, and return a non-zero exit code. I know our users are probably going to be disappointed they didn't get to see the stack trace, but they'll get over it eventually...

Reply:
The exception gets caught in the `main` function, which just prints the error message and gives a non-zero exit code:

```bash
$ cs kill $uuid
Refusing to kill due to duplicate UUIDs.

c3ac6e2d-dc2f-41ed-a3bd-4c2215459869 is duplicated:
- as a job on dev0
- as a job group on dev0

You might need to explicitly set the cluster where you want to kill by using the --cluster flag.
```